src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManager.java

```diff
@@ -980,6 +980,48 @@ public List<GameboardItem> getNextQuestionsForFilter(final GameFilter gameFilter
     return selectionOfGameboardQuestions;
   }
 
+  /**
+   * This generates a set of question in random.
+   *
+   * @param gameFilter - to enable search
+   * @param limit      - to provide a limit of questions
+   * @return a list of questions
+   * @throws ContentManagerException - if there is a problem accessing the content repository.
+   */
+  public List<IsaacQuestionPageDTO> generateRandomQuestions(final GameFilter gameFilter, final int limit)
+      throws ContentManagerException {
+    List<GitContentManager.BooleanSearchClause> fieldsToMap = Lists.newArrayList();
+
+    fieldsToMap.add(new GitContentManager.BooleanSearchClause(TYPE_FIELDNAME, BooleanOperator.AND,
+        Collections.singletonList(QUESTION_TYPE)));
+
+    // exclude questions marked deprecated
+    fieldsToMap.add(new GitContentManager.BooleanSearchClause(DEPRECATED_FIELDNAME, BooleanOperator.NOT,
+        Collections.singletonList("true")));
+
+    fieldsToMap.addAll(generateFieldToMatchForQuestionFilter(gameFilter));
+
+    ResultsWrapper<ContentDTO> results;
+
+    try {
+      results = this.contentManager.findByFieldNamesRandomOrder(fieldsToMap, 0, limit);
+    } catch (ContentManagerException e) {
+      throw e;
+    }
+
+    List<ContentDTO> generatedQuestions = results.getResults();
+    List<IsaacQuestionPageDTO> questionsToReturn = generatedQuestions.stream()
+        .map(question -> (IsaacQuestionPageDTO) question)
+        .filter(qp -> qp.getSupersededBy() == null || qp.getSupersededBy().isEmpty())
+        .collect(Collectors.toList());
+
+    if (questionsToReturn.size() > limit) {
```

Comment:
With the current implementation, this condition will never be true, as you're passing the `limit` to `findByFieldNamesRandomOrder`, so said function will only return a maximum of `limit` objects, never more.